dashboard/src/actions/charts.ts

```diff
@@ -90,15 +94,19 @@ export function fetchCharts(
   repos: string,
   page: number,
   size: number,
+  records: Map<number, boolean>,
   query?: string,
 ): ThunkAction<Promise<void>, IStoreState, null, ChartsAction> {
   return async dispatch => {
-    dispatch(requestCharts(query));
-    try {
-      const charts = await Chart.fetchCharts(cluster, namespace, repos, page, size, query);
-      dispatch(receiveCharts(charts));
-    } catch (e) {
-      dispatch(errorChart(new FetchError(e.message)));
+    const lastPendingPage = Array.from(records.keys()).pop() || 1;
+    if (records.get(page) === false && page <= lastPendingPage) {
```

Comment:
Following the conversation at https://github.com/kubeapps/kubeapps/pull/2264#discussion_r561018190

As a general comment, code should not be added "hoping" to behave somehow, it has to be proven to behave that way with unit tests. I don't see any unit tests enforcing the behavior of this so I'd assume this is not needed. Code that is not "protected" by unit tests is damned to be broken in the future when we don't remember what's doing and we decide to clean it up.

Regarding this issue in particular, I don't think is the duty of the function `fetchCharts` to decide if it should `fetchCharts` or not based on the state. It's the duty of the caller, in this case the `Catalog` component, to not request twice a page (and it's easier to add safeguards and tests there).